### PR TITLE
Fixes #2542 Custom Format on Get Feature Info 

### LINKED ISSUE #2542: Custom Format on Get Feature Info
### Description
Add Markdown Template on Get Feature Info Plugin

- [x] Change layer/group settings from modal to panel
- [x] Add Markdown Template as response format

Due to changes on TOC Settings and Identify Plugins a complete reviews of functionality is needed

### PR DESCRIPTION
**Pull request by allyoucanmap**
## Description

This PR intodruces custom template as feature info response.

## Issues
 - Fix #2542

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)

 - [x] Feature

**What is the current behavior?** (You can also link to an open issue here)
Issue #2542

**What is the new behavior?**
Custom format has been added to default formats

Additional changes:
- Settings as Plugin (removed from TOC toolbar)
- Settings in Dockable Panel
- New container for Identify plugin
- New Icons

New misc components:
- Dock Panel
- Accordion
- Dockable Panel
- Panel Header

**Does this PR introduce a breaking change?** (check one with "x", remove the other)

 - [x] Yes

If this PR contains a breaking change, please describe the impact and migration path for existing applications:

TOCItemsSettings is new plugin for TOC Settings and it needs to be add on localConfig and plugin.js
Identify configuration has changed on some props (localConfig).

**Other information**:

Due to changes on TOC Settings and Identify Plugins a complete reviews of functionality is needed

### CODE REVIEW

File: web/client/components/TOC/TOCItemsSettings.jsx [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+const Toolbar = require('../misc/toolbar/Toolbar');
+const tooltip = require('../misc/enhancers/tooltip');
+const NavItemT = tooltip(NavItem);
+const ResizableModal = require('../misc/ResizableModal');
+const Portal = require('../misc/Portal');
+const {head, isObject} = require('lodash');
+const Message = require('../I18N/Message');
+
+module.exports = props => {
+    const {
```
Comment (mbarto) [inline web/client/components/TOC/TOCItemsSettings.jsx]: This is not generic, too many properties are dependent on specific item info (e.g. chartStyle, getDimension, generalInfoFormat). If it has to be a generic item setting container component, it should only have reference to generic things (e.g. the item node)

File: web/client/components/TOC/TOCItemsSettings.jsx [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+const {Row, Col, Nav, NavItem, Glyphicon} = require('react-bootstrap');
+const DockablePanel = require('../misc/panels/DockablePanel');
+const Toolbar = require('../misc/toolbar/Toolbar');
+const tooltip = require('../misc/enhancers/tooltip');
+const NavItemT = tooltip(NavItem);
+const ResizableModal = require('../misc/ResizableModal');
+const Portal = require('../misc/Portal');
+const {head, isObject} = require('lodash');
+const Message = require('../I18N/Message');
+
```
Comment (mbarto) [inline web/client/components/TOC/TOCItemsSettings.jsx]: Please describe what is this component usable for

File: web/client/components/TOC/enhancers/tocItemsSettings.js [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+            if (!settings.expanded && newProps.settings && newProps.settings.expanded) {
+                // update initial and original settings
+                onUpdateOriginalSettings({...newProps.element});
+                onUpdateInitialSettings({...newProps.element});
+                onSetTab('general');
+            }
+        }
+    })
+);
+
```
Comment (mbarto) [inline web/client/components/TOC/enhancers/tocItemsSettings.js]: Please, describe what you get applying this enhancer to the TOCItemsSettings component

File: web/client/components/TOC/fragments/settings/FeatureInfo.jsx [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+                    panels={data}
+                    onSelect={value => {
+                        const isEqualFormat = this.props.element.featureInfo && this.props.element.featureInfo.format && value === this.props.element.featureInfo.format;
+                        this.props.onChange("featureInfo", {
+                            ...(this.props.element && this.props.element.featureInfo || {}),
+                            format: !isEqualFormat ? value : '',
+                            viewer: this.props.element.featureInfo ? this.props.element.featureInfo.viewer : undefined
+                        });
+                    }}/>
+                <Portal>
```
Comment (mbarto) [inline web/client/components/TOC/fragments/settings/FeatureInfo.jsx]: Every format can have a template? If not this belongs to the format viewer, not to this component

File: web/client/components/data/identify/IdentifyContainer.jsx [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+                        index={index}
+                        setIndex={setIndex}
+                        format={format}
+                        missingResponses={missingResponses}
+                        responses={responses}
+                        {...viewerOptions}/>
+            </DockablePanel>
+            {enableRevGeocode &&
+            // geocode modal
+            <Portal>
```
Comment (mbarto) [inline web/client/components/data/identify/IdentifyContainer.jsx]: This should be an autonomous component

File: web/client/components/data/identify/enhancers/defaultViewer.js [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+    )
+);
+
+const defaultViewerDefaultProps = defaultProps({
+    format: MapInfoUtils.getDefaultInfoFormatValue(),
+    validator: MapInfoUtils.getValidator
+});
+
+module.exports = {
+    defaultViewerHanlders,
```
Comment (mbarto) [inline web/client/components/data/identify/enhancers/defaultViewer.js]: handlers

File: web/client/components/data/identify/enhancers/defaultViewer.js [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+    withState(
+        'index', 'setIndex', 0
+    )
+);
+
+const defaultViewerDefaultProps = defaultProps({
+    format: MapInfoUtils.getDefaultInfoFormatValue(),
+    validator: MapInfoUtils.getValidator
+});
+
```
Comment (mbarto) [inline web/client/components/data/identify/enhancers/defaultViewer.js]: Please, describe how you can use these enhancers

File: web/client/components/data/identify/enhancers/identify.js [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+                purgeResults();
+            }
+            // reset current page on new requests set
+            if (setIndex && !isEqual(newProps.responses, responses)) {
+                setIndex(0);
+            }
+        }
+    })
+);
+
```
Comment (mbarto) [inline web/client/components/data/identify/enhancers/identify.js]: Please, describe how you can use these enhancers

File: web/client/components/data/identify/viewers/CustomViewer.jsx [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+ * LICENSE file in the root directory of this source tree.
+ */
+
+const React = require('react');
+const {template} = require('lodash');
+const MapInfoUtils = require('../../../../utils/MapInfoUtils');
+const HtmlRenderer = require('../../../misc/HtmlRenderer');
+const {Row, Col, Grid} = require('react-bootstrap');
+
+module.exports = ({layer = {}, response}) => (
```
Comment (mbarto) [inline web/client/components/data/identify/viewers/CustomViewer.jsx]: Maybe TemplateViewer is better than CustomViewer

File: web/client/components/data/identify/viewers/__tests__/viewers-test.jsx [hunk 10]
```diff
@@ -87,4 +87,205 @@ describe('Identity Viewers', () => {
...
         expect(cmpDom.innerText.indexOf('This is my viewer: 1') !== -1).toBe(true);
     });
+
+    it('test JSONViewer with Markdown', () => {
+        const cmp = ReactDOM.render(
+        <JSONViewer
+            layer={{
+                featureInfo: {
+                    format: 'CUSTOM',
+                    template: '<p id="my-custom-markdown">the property name is ${ properties.name }</p>'
```
Comment (mbarto) [inline web/client/components/data/identify/viewers/__tests__/viewers-test.jsx]: This doesn't look like Markdown to me

File: web/client/components/TOC/TOCItemsSettings.jsx [hunk 11]
```diff
@@ -0,0 +1,144 @@
...
+/**
+ * Component for rendering TOC Settings as tabs inside a Dockable contanier
+ * @memberof components.TOC
+ * @name TOCItemsSettings
+ * @class
+ * @prop {dock} dock switch between Dockable Panel and Resizable Modal, default true (DockPanel)
```
Comment (offtherailz) [inline web/client/components/TOC/TOCItemsSettings.jsx]: probably `{boolean}`

File: web/client/components/TOC/Toolbar.jsx [hunk 12]
```diff
@@ -12,8 +12,9 @@ const {ButtonGroup, Button, Glyphicon, Tooltip, OverlayTrigger} = require('react
 const ReactCSSTransitionGroup = require('react-addons-css-transition-group');
 const {head} = require('lodash');
 const ConfirmModal = require('../maps/modals/ConfirmModal');
-const SettingsModal = require('./fragments/SettingsModal');
-const GroupSettingsModal = require('./fragments/GroupSettingsModal');
+// removed from toolbar to avoid conflict with TOCItemsSettings
+// const SettingsModal = require('./fragments/SettingsModal');
```
Comment (offtherailz) [inline web/client/components/TOC/Toolbar.jsx]: remove commented code from this file

**Comment by offtherailz** [comment]
Noticed this:
image
Embedded and Mobile seems to have some problems (at least in custom projects)

**Pull request merged**